app/models/hardware.rb

```diff
@@ -44,6 +43,11 @@ def mac_addresses
   def ram_size_in_bytes
     memory_mb.to_i * 1.megabyte
   end
+  # resulting sql: "(CAST("hardwares"."memory_mb" AS bigint) * 1048576)"
+  virtual_attribute :ram_size_in_bytes, :integer, :arel => (lambda do |t|
+    t.grouping(Arel::Nodes::Multiplication.new(
+      Arel::Nodes::NamedFunction.new("CAST", [t[:memory_mb].as("bigint")]), 1.megabyte))
+  end)
```

Comment:
I was testing before/after manually.
Not sure how to write rspec,  ```virtual_column_sql_value(Hardware, "ram_size_in_bytes")```  is working fine.  Also, ```ActiveRecord::Base.connection.execute("SELECT hardwares.memory_mb * 1048576 FROM hardwares")``` failing on console :
<img width="814" alt="screen shot 2017-07-12 at 3 44 19 pm" src="https://user-images.githubusercontent.com/6556758/28136792-03ff947c-6719-11e7-976e-3f583754a843.png">
but works fine from rspec